### PR TITLE
etcdserver: separate out raft related stuff

### PR DESCRIPTION
**Pull request by xiang90**
@yichengq 
This is the first step to separate out raft related logic from server. 
The goal of this commit is to move out the variables and independent functions to raft.

### CODE REVIEW

File: etcdserver/server.go [hunk 1]
```diff
@@ -122,7 +120,12 @@ type RaftTimer interface {
 
 // EtcdServer is the production implementation of the Server interface
 type EtcdServer struct {
-	cfg        *ServerConfig
+	cfg *ServerConfig
+
+	r raftNode
+
+	// TODO: move ticker into raftNode
+	Ticker     <-chan time.Time
```
Comment (yichengq) [inline etcdserver/server.go]: remove this one?

Comment (xiang90) [inline etcdserver/server.go]: done

File: etcdserver/raft.go [hunk 2]
```diff
@@ -19,16 +19,107 @@ package etcdserver
...
+}
+
+type raftNode struct {
+	raft.Node
+
+	// config
+	snapCount uint64 // number of entries to trigger a snapshot
+
+	// utility
+	Ticker      <-chan time.Time
```
Comment (yichengq) [inline etcdserver/raft.go]: Ticker -> ticker

**Comment by yichengq** [comment]
lgtm

**Pull request merged**